src/robotFactory.js

```diff
@@ -1,10 +1,75 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    const {
+      x = 0,
+      y = 0,
+    } = coords;
 
-class DeliveryDrone {}
+    this.coords = {
+      x, y,
+    };
```

Comment:
```suggestion
    this.coords = {
      x: coords.x || 0,
      y: coords.y || 0,
    };
```
You can set them right here without destructuring